psutil/_psutil_osx.c

```diff
@@ -279,16 +280,24 @@ static PyObject *
 psutil_proc_pidtaskinfo_oneshot(PyObject *self, PyObject *args) {
     pid_t pid;
     struct proc_taskinfo pti;
+    struct mach_timebase_info info;
+    uint64_t total_user, total_system;
 
     if (! PyArg_ParseTuple(args, _Py_PARSE_PID, &pid))
         return NULL;
     if (psutil_proc_pidinfo(pid, PROC_PIDTASKINFO, 0, &pti, sizeof(pti)) <= 0)
         return NULL;
 
+    mach_timebase_info(&info);
+    total_user = pti.pti_total_user * info.numer;
+    total_user /= info.denom;
+    total_system = pti.pti_total_system * info.numer;
+    total_system /= info.denom;
+
     return Py_BuildValue(
         "(ddKKkkkk)",
-        (float)pti.pti_total_user / 1000000000.0,     // (float) cpu user time
-        (float)pti.pti_total_system / 1000000000.0,   // (float) cpu sys time
+        (float)total_user / 1000000000.0,     // (float) cpu user time
+        (float)total_system / 1000000000.0,   // (float) cpu sys time
```

Comment:
Since you did the new conversion above, is the `/ 1000000000.0` division still necessary?

Reply:
Yes. The mach timebase conversion ratio produces nanoseconds.